src/Context.php

```diff
@@ -16,6 +16,9 @@ class Context
     /** @var string */
     private $signingKeyId;
 
+    /** @var bool */
+    private $signatureHeadersSpecified;
```

Comment:
I can't see where this is ever set.

But I don't understand why you need it - you already have `$this->headers` which should indicate whether it has been set or not

Reply:
Good point, guess I was stuck on the old approach and just ported it to calling functions. Works fine without this tracker. e8d9b91904f43f80af78bca6cacbc72208669a06